lib/streamManager.js

```diff
@@ -177,16 +177,20 @@ var exports = module.exports = (function () {
             }));
         },
 
-        flushFunction: function () {
-            end = true;
+        flushFunction: function (cb) {
+            end = cb;
             if (unprocessed === 0) {
-                this.emit('end');
+                return cb();
             }
         },
 
         additionalStreams: function () {
             var options = exports.options;
 
+            additionalFiles = [];
+            unprocessed = 0;
+            end = undefined;
```

Comment:
You can repro by forcing a test to fail and seeing it's effect on subsequent tests.  For example, if you update 'should work with relative paths' to force a failure:

````javascript
// approx line 576 in test.js
        stream.on('data', function(newFile){
            should.exist(newFile.contents);
            if (a === 1) {
                'a'.should.equal('b');  // Force a failure
                path.normalize(newFile.path).should.equal(path.normalize('./test/fixtures/scripts/combined.js'));
            }
            ++a;
        });
````

You'll see that subsequent tests all time out